hudi-spark/src/main/java/org/apache/hudi/keygen/KeyGenerator.java

```diff
@@ -40,4 +40,22 @@ protected KeyGenerator(TypedProperties config) {
    * Generate a Hoodie Key out of provided generic record.
    */
   public abstract HoodieKey getKey(GenericRecord record);
+
+  public abstract String getPartitionPath(GenericRecord record, String partitionPathField);
+
+  public abstract String getRecordKey(GenericRecord record);
+
+  public enum PartitionKeyType {
```

Comment:
ah, you have defined an enum. May I know why using strings back there ?

Reply:
added a function to use it now.